src/main/java/com/altigee/needoff/auth/service/AccountServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.altigee.needoff.auth.service;
+
+import com.altigee.needoff.auth.data.AccountRepo;
+import com.altigee.needoff.auth.data.RoleRepo;
+import com.altigee.needoff.auth.model.Account;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class AccountServiceImpl implements AccountService {
+
+  @Autowired private AccountRepo accountRepo;
+
+  @Autowired private RoleRepo roleRepo;
+
+  @Override
+  public Optional<Account> grantAccountAdminRoles(Long accountId) {
+
+    return accountRepo
+        .findById(accountId)
+        .map(
```

Comment:
You can use `Stream::peek`, like so `.findById(accountId).peek(account -> account.setRoles(...))...`

Reply:
its `Optional.map` which doesn't have `peek` function